apps/user-profile/src/app/pages/settings-page/settings-authorization/settings-authentication.component.ts

```diff
@@ -16,115 +20,126 @@ import { MatSlideToggle } from '@angular/material/slide-toggle';
 export class SettingsAuthenticationComponent implements OnInit {
   @ViewChild('toggle') toggle: MatSlideToggle;
 
-  removeDialogTitle: string;
   imgAtt: Attribute;
+  textAtt: Attribute;
   imageSrc = '';
-  removeDialogDescription: string;
   mfaUrl = '';
   displayImageBlock: boolean;
-  loadingImg = false;
-  saveImgSuccess = '';
-  removeImgSuccess = '';
+  displayTextBlock: boolean;
+  loading = false;
 
   constructor(
     private dialog: MatDialog,
     private attributesManagerService: AttributesManagerService,
     private store: StoreService,
-    private translate: TranslateService,
+    private translate: PerunTranslateService,
     private notificatorService: NotificatorService
-  ) {
-    translate
-      .get('AUTHENTICATION.DELETE_IMG_DIALOG_TITLE')
-      .subscribe((res: string) => (this.removeDialogTitle = res));
-    translate
-      .get('AUTHENTICATION.DELETE_IMG_DIALOG_DESC')
-      .subscribe((res: string) => (this.removeDialogDescription = res));
-    translate
-      .get('AUTHENTICATION.SAVE_IMG_SUCCESS')
-      .subscribe((res: string) => (this.saveImgSuccess = res));
-    translate
-      .get('AUTHENTICATION.REMOVE_IMG_SUCCESS')
-      .subscribe((res: string) => (this.removeImgSuccess = res));
-  }
+  ) {}
 
   ngOnInit(): void {
     const mfa = this.store.getProperty('mfa');
 
     this.translate.onLangChange.subscribe(() => {
-      this.translate
-        .get('AUTHENTICATION.DELETE_IMG_DIALOG_TITLE')
-        .subscribe((res: string) => (this.removeDialogTitle = res));
-      this.translate
-        .get('AUTHENTICATION.DELETE_IMG_DIALOG_DESC')
-        .subscribe((res: string) => (this.removeDialogDescription = res));
       this.mfaUrl = this.translate.currentLang === 'en' ? mfa.url_en : mfa.url_cs;
     });
 
     this.mfaUrl = this.translate.currentLang === 'en' ? mfa.url_en : mfa.url_cs;
     this.displayImageBlock = this.store.getProperty('mfa').enable_security_image;
     if (this.displayImageBlock) {
-      this.loadImage();
+      this.loadSecurityAttribute('image');
+    }
+    this.displayTextBlock = this.store.getProperty('mfa').enable_security_text;
+    if (this.displayTextBlock) {
+      this.loadSecurityAttribute('text');
     }
   }
 
-  loadImage(): void {
-    this.loadingImg = true;
-    const imgAttributeName = this.store.getProperty('mfa').security_image_attribute;
+  loadSecurityAttribute(mode: 'image' | 'text'): void {
```

Comment:
Using this "mode" argument throughout the component is in my opinion not the best solution. It seems like every other line is a `if else` statment. If at some point we would decine to allow another "mode" we need to add a lot of new branches so it is not easily expandable.

If possible pass all required information as an argument as it is known by the "caller".

Reply:
I see your point, but IMO some if statements would still need to stay (eg. assigning the attribute to imgAttr or textAttr) and overall the mode argument makes the methods easier to use, no? Maybe a switch, which would set the variables based on mode at the beginning of the method would make it easier expandable, should a new mode be added?

Reply:
Yes, perhaps some if statements are neccessery which is fine but right now it feels like each assigment is a if else. It only looks somewhat clean because of the `? :` operator. For the example of assigning attributes I just imagined something like `imgAttr = load(attrName);`.
I think the mode argument passes all logic handling to the function itself and that I think is a bad "design". Imagine we have 10 modes instead of 2 or 3 now each method starts with a 10 case switch that just sets the rights values. Instead I would like to pass the correct values as arguments since they are known by the caller. You can still get some suggestions/autocomplete from IDE as to what is a valid input by giving it a nice type not just e.g. string. Now instead of adding new case to multiple methods you just need to expand the input type.

Reply:
I am not sure if it looks better now since the `loadSecurityAttribute(...)` method needs to be asynchronous, but you can consider it on your own. Maybe there would be helpful some rxjs magic, but I wasn't able to find the better solution in rxjs.

Reply:
I think it is better and the amount of rxjs magic is just right. To touch on the number of modes we will probalby support I don't necesserily excpet we will add more modes or that some other ones would even make sense. I just wanted to point out that in my opinion this usage of "mode" is just digging a hole for ourselfs. The functions handle a lot of unnecessary logic which makes them harder to read and expand. It also leads to more usage of the mode as it leaks to other parts of the code and its a whole cycle then.